addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -97,12 +97,17 @@ function factory(dependencies) {
          * Remove this attachment globally.
          */
         async remove() {
+            if (this.isUnlinkPending) {
+                return;
+            }
             if (!this.isTemporary) {
+                this.isUnlinkPending = true;
                 await this.async(() => this.env.services.rpc({
                     model: 'ir.attachment',
                     method: 'unlink',
                     args: [this.id],
                 }, { shadow: true }));
+                this.isUnlinkPending = undefined;
```

Comment:
if the unlink rpc has an issue (server failure or whatever) the value will never be reset and it will not be possible to delete the attachment until reload

should try/finally the rpc to ensure the value will be reset

also as it is a boolean, I would suggest have just true/false and not undefined